tests/test_utils.py

```diff
@@ -374,6 +374,39 @@ def _second_callback(result):
     assert second_results == [3]
 
 
+@pytest.mark.asyncio
+async def test_concurrent_runner_try_put():
+    results = []
+
+    def _results_callback(result):
+        results.append(result)
+
+    async def coroutine(i):
+        await asyncio.sleep(0.1)
+        return i
+
+    runner = ConcurrentTasks(1, results_callback=_results_callback)
```

Comment:
This is an absolute nit, but I could see some benefit in having a test with a higher ConcurrentTasks limit. Right now there's no test to confirm that `try_put` will work if the pool is 1/2 full, for example.

Reply:
Updated: https://github.com/elastic/connectors/pull/1479/commits/63675a9ed825ad8def6abd2077376ed24b936d26

Reply:
I think I accidentally forgot to remove those code: https://github.com/elastic/connectors/pull/1479/commits/0e539b01de8fdb9cf1bcf6f74567a70e9e0d60c4